ui/js/components/auth.js

```diff
@@ -0,0 +1,165 @@
+"use strict";
+
+/**
+ * This component handles logging in to Taskcluster Authentication
+ *
+ * See: https://docs.taskcluster.net/manual/3rdparty
+ *
+ * This communicates to the rest of Treeherder by calling the onUserChange
+ * function on the HTML element, which in turn stores that user in $rootScope.
+ */
+treeherder.component("login", {
+    template: `
+        <span class="dropdown"
+              ng-if="user.loggedin">
+          <button id="logoutLabel" title="Logged in as: {{user.email}}" role="button"
+                  data-toggle="dropdown" data-target="#"
+                  class="btn btn-view-nav btn-right-navbar nav-persona-btn">
+            <div class="nav-user-icon">
+              <span class="fa fa-user pull-left"></span>
+            </div>
+            <span class="fa fa-angle-down lightgray"></span>
+          </button>
+          <ul class="dropdown-menu" role="menu" aria-labelledby="logoutLabel">
+              <li>
+                <a ng-click="logout()">Logout</a>
+              </li>
+          </ul>
+        </span>
+
+        <a class="btn btn-view-nav btn-right-navbar nav-login-btn"
+           ng-if="!user.loggedin"
+           ng-click="login()">Login/Register</a>
+    `,
+    bindings: {
+        onUserChange: "&"
+    },
+    controller: ['$scope', '$location', '$window', 'localStorageService',
+        'ThUserModel', '$http', 'thUrl', '$timeout',
+        function ($scope, $location, $window, localStorageService,
+                  ThUserModel, $http, thUrl, $timeout) {
```

Comment:
I don't think you need to use $scope with new-style angular components, instead I think you can just attach things to `this`.  See `ui/js/components/perf/compare.js` for some examples.